pkg/controllers/state/node.go

```diff
@@ -106,7 +106,13 @@ func NewNode() *Node {
 }
 
 func (in *Node) Name() string {
```

Comment:
Can this be called something different than `Node`? There's a lot of awkward stuttering in some other places (`cn.node.node`). It's also not an actual node, just a different representation of a node. Maybe `CachedNode`?

Reply:
I don't really like `CachedNode` either. I went with `StateNode`. I know this stutters but it stutters a little less than `cn.Node.Node` :)